### ISSUE TITLE
Use xxHash instead of MurmurHash3

### ISSUE THREAD

**Issue by plokhotnyuk** [alexandrnikitin/bloom-filter-scala#7]
http://cyan4973.github.io/xxHash/

**Comment by alexandrnikitin** [comment]
Agree, it would be nice to have it built into the lib. I found a good implementation here: https://github.com/OpenHFT/Zero-Allocation-Hashing

**Comment by velvia** [comment]
+1
Although this can be worked around just by using XXHash itself to compute a 64-bit value.

Here is another XXHash implementation which seems fairly fast:
https://github.com/jpountz/lz4-java

**Comment by asoltysik** [comment]
How is the uniformity of xxHash? Couldn't find any bechmarks wrt to that.

**Comment by velvia** [comment]
According to the XXHash website, it has a very high uniformity score.  However testing is best.
